tsdb/batcher.go

```diff
@@ -55,14 +55,23 @@ func (b *PointBatcher) Start() {
 		return
 	}
 
-	var timer *time.Timer
+	// initialize the timer variable
+	timer := &time.NewTimer(b.duration)
+	timer.Stop()
 	var batch []models.Point
-	var timerCh <-chan time.Time
 
 	emit := func() {
+
+		timer.Stop()
```

Comment:
We are reusing a timer with `Reset`, so we need to handle the case where we've reached this point other than the `<-timer.C` case, but the timer fired concurrently with this function call. That means that stop wasn't effective, and the `<-timer.C` case will fire immediately once we loop around again. This won't matter much since `batch` will be empty, and it will just return after draining the channel. But we should probably avoid the extra call if we can:

```go
timer.Stop()
select {
    case <-timer.C:
    default:
}
```